synapse/rest/media/v1/media_repository.py

```diff
@@ -150,6 +155,98 @@ def mark_recently_accessed(self, server_name: Optional[str], media_id: str) -> N
         else:
             self.recently_accessed_locals.add(media_id)
 
+    async def create_media_id(self, auth_user: UserID) -> Tuple[str, int]:
+        """Create and store a media ID for a local user and return the mxc URL
+
+        Args:
+            auth_user: The user_id of the uploader
+
+        Returns:
+            The mxc url of the stored content
+        """
+        media_id = random_string(24)
+        now = self.clock.time_msec()
+        unused_expires_at = now + (60 * 1000)
```

Comment:
I think this needs to be configurable (especially for tests). What section of the config file should this be in?